scrapy/utils/datatypes.py

```diff
@@ -64,21 +65,57 @@ def pop(self, key, *args):
 
 
 class LocalCache(collections.OrderedDict):
-    """Dictionary with a finite number of keys.
-
-    Older items expires first.
+    """
+    Dictionary with a finite number of keys.
+    Older items expire first.
     """
 
-    def __init__(self, limit=None):
+    def __init__(self, limit=None, time_limit=None):
         super().__init__()
         self.limit = limit
+        self.time_limit = time_limit
 
     def __setitem__(self, key, value):
-        if self.limit:
+        now = datetime.now()
+        value = (value, now)
+
+        if self.limit is not None:
             while len(self) >= self.limit:
                 self.popitem(last=False)
+
+        while True:
+            try:
+                val = self.values().__iter__().__next__()
```

Comment:
```suggestion
                val = next(iter(self.values()))
```
Same in the other occurrence below.